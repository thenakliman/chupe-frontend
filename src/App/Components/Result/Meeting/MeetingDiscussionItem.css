.meeting-discussion-item-container {
  margin: 1rem 18rem;
}

.meeting-discussion-item-button-container {
  display: flex;
  justify-content: end;
  height: 2.5rem;
  margin-bottom: 1rem;
}

.meeting-discussion-item-button {
  cursor: pointer;
}