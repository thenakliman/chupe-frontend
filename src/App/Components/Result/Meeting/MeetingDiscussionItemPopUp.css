.meeting-discussion-item-pop-up-container {
  position: fixed;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  margin: auto;
  background-color: rgba(0, 0, 0, 0.5);
}

.meeting-discussion-item-pop-up-inner {
  position: absolute;
  left: 20rem;
  top: 10rem;
  margin: auto;
  background: white;
  padding: 3rem;
  display: flex;
  flex-direction: column;
}

.button-container-meeting-discussion-item {
  display: flex;
  flex-direction: row;
  margin-bottom: 2rem;
}

.meeting-discussion-item-buttons {
  padding: 0.8rem 1.2rem;
  margin-right: 5rem;
  font-size: 1rem;
  cursor: pointer;
}

.meeting-discussion-item {
  border: 1px solid rgb(180, 180, 180);
  width: 45rem;
  height: 10rem;
  word-wrap: break-word;
  padding: 0.5rem;
  margin-bottom: 2rem;
}

.select-meeting-discussion-item-pop-up {
  padding: 0.6rem 1.5rem;
  width: 15rem;
  margin-bottom: 2rem;
}