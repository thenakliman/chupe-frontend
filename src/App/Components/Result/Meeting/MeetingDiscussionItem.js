import React from 'react';
import propTypes from 'prop-types';
/* eslint-disable */
import {MeetingDiscussionItemPopUp} from './MeetingDiscussionItemPopUp';
/* eslint-enable */

require('./MeetingDiscussionItem.css');

export class MeetingDiscussionItem extends React.Component {
  constructor(props) {
    super(props);
    this.state = {isAddingMeetingDiscussionItem: false};
    this.toggleMeetingDiscussionItemPopUp =
        this.toggleMeetingDiscussionItemPopUp.bind(this);
    this.onSubmitOfMeetingDiscussionItemPopUp =
        this.onSubmitOfMeetingDiscussionItemPopUp.bind(this);
  }

  componentDidMount() {
    this.props.getMeetingDiscussionItems(
        this.props.match.params.id);

    this.props.getAllUsers();
  }

  toggleMeetingDiscussionItemPopUp() {
    this.setState({
      isAddingMeetingDiscussionItem:
          !this.state.isAddingMeetingDiscussionItem,
    });
  }

  onSubmitOfMeetingDiscussionItemPopUp(meetingDiscussionItem) {
    this.props.createMeetingDiscussionItem({
      ...meetingDiscussionItem,
      meetingId: this.props.match.params.id,
    });
    this.toggleMeetingDiscussionItemPopUp();
  }

  render() {
    return (
        <div className={'meeting-discussion-item-container'}>
          {this.state.isAddingMeetingDiscussionItem &&
          <MeetingDiscussionItemPopUp
              users={this.props.users}
              onSubmit={this.onSubmitOfMeetingDiscussionItemPopUp}
              onCancel={this.toggleMeetingDiscussionItemPopUp}
          />
          }
          <div className={'meeting-discussion-item-button-container'}>
            <button id='meeting-discussion-item-button'
                    className={'meeting-discussion-item-button'}
                    onClick={() => this.toggleMeetingDiscussionItemPopUp()}>
              Create Meeting Discussion Item
            </button>
          </div>
          <table id='all-meeting-discussion-items-ordered-list'>
            <thead>
            <tr className='meeting-discussion-item-table-header'>
              <th> Id</th>
              <th> Subject</th>
              <th> Assigned To</th>
              <th> Created By</th>
            </tr>
            </thead>
            <tbody id='all-meeting-discussion-item-table-body-id'
                   className={'meeting-discussion-item-table-body'}>
            {
              this.props.meetingDiscussionItems.map((mDiscussionItem) => (
                  <tr key={`${mDiscussionItem.id}`}>
                    <td>{mDiscussionItem.id}</td>
                    <td>{mDiscussionItem.discussionItem}</td>
                    <td>{mDiscussionItem.assignedTo}</td>
                    <td>{mDiscussionItem.createdBy}</td>
                  </tr>
              ))
            }
            </tbody>
          </table>
        </div>
    );
  }
}

MeetingDiscussionItem.propTypes = {
  getMeetingDiscussionItems: propTypes.func.isRequired,
  getAllUsers: propTypes.func.isRequired,
  createMeetingDiscussionItem: propTypes.func.isRequired,
  meetingDiscussionItems: propTypes.array.isRequired,
  users: propTypes.array.isRequired,
};
