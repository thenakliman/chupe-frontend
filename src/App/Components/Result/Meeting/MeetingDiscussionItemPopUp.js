import React from 'react';
import propTypes from 'prop-types';

require('./MeetingDiscussionItemPopUp.css');

export class MeetingDiscussionItemPopUp extends React.Component {
  constructor(props) {
    super(props);
    this.state = {discussionItem: '', discussionItemType: '', assignedTo: ''};
    this.handleSubjectChange = this.handleSubjectChange.bind(this);
    this.handleMeetingDiscussionItemTypeChange = this.handleMeetingDiscussionItemTypeChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  handleSubjectChange(value) {
    this.setState({discussionItem: value});
  }

  handleMeetingDiscussionItemTypeChange(value) {
    this.setState({discussionItemType: value});
  }

  handleAssignedToUserChange(value) {
    this.setState({assignedTo: value});
  }

  onSubmit() {
    this.props.onSubmit({
        discussionItem: this.state.discussionItem,
        discussionItemType: this.state.discussionItemType,
        assignedTo: this.state.assignedTo,
    });
  }

  render() {
    return (
      <div className={'meeting-discussion-item-pop-up-container'}>
      <div className={'meeting-discussion-item-pop-up-inner'}>
        <div>
          Meeting Discussion Item:
        </div>
          <textarea className={'meeting-discussion-item'}
                  value={this.state.discussionItem}
                  onChange={(event) => this.handleSubjectChange(event.target.value)}
                  resize={'none'}/>

        Assigned To
        <select value={this.state.assignedTo}
                id={'select-assigned-to'}
                className={'select-meeting-discussion-item-pop-up'}
                onChange={(event) => this.handleAssignedToUserChange(event.target.value)}
        >
        <option key={''} value={''}> Select user </option>
        {
          this.props.users.map((user) =>
            <option key={user.userName} value={user.userName}>
              {user.userName}
            </option>)
        }
        </select>
        Meeting Discussion Item Type:
        <select value={this.state.discussionItemType}
                id={'select-discussion-item-type'}
                className={'select-meeting-discussion-item-pop-up'}
                onChange={(event) => this.handleMeetingDiscussionItemTypeChange(event.target.value)}
        >
        <option key={''} value={''}> Select Discussion Item type </option>
        <option key={'INFORMATION'} value={'INFORMATION'}> Information </option>
        <option key={'ACTION_ITEM'} value={'ACTION_ITEM'}> Action Item </option>
        </select>
        <div className={'button-container-meeting-discussion-item'}>
          <button className={'meeting-discussion-item-buttons'}
                  id={'meeting-discussion-item-submit-button'}
                  onClick={this.onSubmit}>
              Submit
          </button>
          <button className={'meeting-discussion-item-buttons'}
                  id={'meeting-discussion-item-cancel-button'} onClick={this.props.onCancel}>
              Cancel
          </button>
        </div>
      </div>
      </div>
    );
  }
}

MeetingDiscussionItemPopUp.propTypes = {
  users: propTypes.array.isRequired,
  onSubmit: propTypes.func.isRequired,
  onCancel: propTypes.func.isRequired,
};
