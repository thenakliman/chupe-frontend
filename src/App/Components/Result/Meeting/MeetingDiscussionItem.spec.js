/* eslint-disable */
import React from 'react';
import {MeetingDiscussionItem} from './MeetingDiscussionItem';
import {MeetingDiscussionItemPopUp} from './MeetingDiscussionItemPopUp';
/* eslint-enable */
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';


const meetingDiscussionItems = [{
  id: 10101,
  subject: 'meeting - 1',
}, {
  id: 10102,
  subject: 'meeting - 2',
}];

describe('Show meeting discussion item component', () => {
  describe('should match snapshot', () => {
    const wrapper = shallow(<MeetingDiscussionItem
       match={{params: {id: 101}}}
       meetingDiscussionItems={meetingDiscussionItems}
       getMeetingDiscussionItems={jest.fn()}
       getAllUsers={jest.fn()}
       users={[]}
       createMeetingDiscussionItem={jest.fn()}
    />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call getMeetingDiscussionItems on component did mount ', () => {
    const getMeetingDiscussionItems = jest.fn();
    const meetingId = 101;
    shallow(
        <MeetingDiscussionItem
          match={{params: {id: meetingId}}}
          meetingDiscussionItems={meetingDiscussionItems}
          getMeetingDiscussionItems={getMeetingDiscussionItems}
          getAllUsers={jest.fn()}
          users={[]}
          createMeetingDiscussionItem={jest.fn()}
        />
    );

    expect(getMeetingDiscussionItems).toHaveBeenCalledWith(meetingId);
  });

  it('should show row for each meetingDiscussionItems', () => {
    const getMeetingDiscussionItems = jest.fn();
    const meetingId = 101;
    const wrapper = shallow(
        <MeetingDiscussionItem
          match={{params: {id: meetingId}}}
          meetingDiscussionItems={meetingDiscussionItems}
          getMeetingDiscussionItems={getMeetingDiscussionItems}
          getAllUsers={jest.fn()}
          users={[]}
          createMeetingDiscussionItem={jest.fn()}
        />
    );

    expect(wrapper.find('tr').length).toBe(3);
  });

  it('should have initial state isAddingMeetingDiscussionItem set to false', () => {
    const getMeetingDiscussionItems = jest.fn();
    const meetingId = 101;
    const wrapper = shallow(
        <MeetingDiscussionItem
          match={{params: {id: meetingId}}}
          meetingDiscussionItems={meetingDiscussionItems}
          getMeetingDiscussionItems={getMeetingDiscussionItems}
          getAllUsers={jest.fn()}
          users={[]}
          createMeetingDiscussionItem={jest.fn()}
        />
    );

    expect(wrapper.state().isAddingMeetingDiscussionItem).toBeFalsy();
  });

  it('should call get all users on component did mount', () => {
    const getMeetingDiscussionItems = jest.fn();
    const meetingId = 101;
    const getAllUsers = jest.fn();
    const wrapper = shallow(
        <MeetingDiscussionItem
          match={{params: {id: meetingId}}}
          meetingDiscussionItems={meetingDiscussionItems}
          getMeetingDiscussionItems={getMeetingDiscussionItems}
          users={[]}
          getAllUsers={getAllUsers}
          createMeetingDiscussionItem={jest.fn()}
        />
    );

    expect(getAllUsers).toHaveBeenCalledWith();
  });

  it('should set give meeting discussion item pop up to true on click of button', () => {
    const getMeetingDiscussionItems = jest.fn();
    const meetingId = 101;
    const createFeedback = jest.fn();
    const wrapper = shallow(
        <MeetingDiscussionItem
          match={{params: {id: meetingId}}}
          meetingDiscussionItems={meetingDiscussionItems}
          getMeetingDiscussionItems={getMeetingDiscussionItems}
          getAllUsers={jest.fn()}
          createMeetingDiscussionItem={createFeedback}
        />
    );

    wrapper.find('#meeting-discussion-item-button').simulate('click');

    expect(wrapper.state().isAddingMeetingDiscussionItem).toBeTruthy();
  });

  it('should show give meeting discussion item pop up when isAddingMeetingDiscussionItem is true', () => {
    const getMeetingDiscussionItems = jest.fn();
    const meetingId = 101;
    const createFeedback = jest.fn();
    const wrapper = shallow(
        <MeetingDiscussionItem
          match={{params: {id: meetingId}}}
          meetingDiscussionItems={meetingDiscussionItems}
          getMeetingDiscussionItems={getMeetingDiscussionItems}
          getAllUsers={jest.fn()}
          createMeetingDiscussionItem={createFeedback}
        />
    );

    wrapper.find('#meeting-discussion-item-button').simulate('click');

    expect(wrapper.find(MeetingDiscussionItemPopUp).length).toBe(1);
  });

  it('should provide users as property to MeetingDiscussionItemPopUp', () => {
    const meetingId = 101;
    const users = [{id: 10}];
    const wrapper = shallow(
        <MeetingDiscussionItem
          match={{params: {id: meetingId}}}
          meetingDiscussionItems={[]}
          users={users}
          getMeetingDiscussionItems={jest.fn()}
          getAllUsers={jest.fn()}
          createMeetingDiscussionItem={jest.fn()}
        />
    );

    wrapper.setState({isAddingMeetingDiscussionItem: true});

    expect(wrapper.find(MeetingDiscussionItemPopUp).props().users).toBe(users);
  });

  it('should change pop up on call of onCancel of MeetingDiscussionItemPopUp', () => {
    const meetingId = 101;
    const users = [{id: 10}];
    const wrapper = shallow(
        <MeetingDiscussionItem
          match={{params: {id: meetingId}}}
          meetingDiscussionItems={[]}
          users={users}
          getMeetingDiscussionItems={jest.fn()}
          getAllUsers={jest.fn()}
          createMeetingDiscussionItem={jest.fn()}
        />
    );

    wrapper.setState({isAddingMeetingDiscussionItem: true});
    wrapper.find(MeetingDiscussionItemPopUp).props().onCancel();

    expect(wrapper.state().isAddingMeetingDiscussionItem).toBeFalsy();
  });

  it('should call create meeting discussion item on call of onSubmit of pop up', () => {
    const meetingId = 101;
    const users = [{id: 10}];
    const createMeetingDiscussionItem = jest.fn();
    const wrapper = shallow(
        <MeetingDiscussionItem
          match={{params: {id: meetingId}}}
          meetingDiscussionItems={[]}
          users={users}
          getMeetingDiscussionItems={jest.fn()}
          getAllUsers={jest.fn()}
          createMeetingDiscussionItem={createMeetingDiscussionItem}
        />
    );

    wrapper.setState({isAddingMeetingDiscussionItem: true});
    const meetingDiscussionItem = {
        subject: 'subject data',
    };

    wrapper.find(MeetingDiscussionItemPopUp).props().onSubmit(meetingDiscussionItem);
    expect(createMeetingDiscussionItem).toHaveBeenCalledWith(
        {...meetingDiscussionItem, meetingId: meetingId});
  });
});
