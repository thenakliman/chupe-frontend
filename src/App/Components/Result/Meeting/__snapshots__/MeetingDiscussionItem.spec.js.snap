// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="meeting-discussion-item-container"
>
  <div
    className="meeting-discussion-item-button-container"
  >
    <button
      className="meeting-discussion-item-button"
      id="meeting-discussion-item-button"
      onClick={[Function]}
    >
      Create Meeting Discussion Item
    </button>
  </div>
  <table
    id="all-meeting-discussion-items-ordered-list"
  >
    <thead>
      <tr
        className="meeting-discussion-item-table-header"
      >
        <th>
           Id 
        </th>
        <th>
           Subject 
        </th>
        <th>
           Assigned To 
        </th>
        <th>
           Created By 
        </th>
      </tr>
    </thead>
    <tbody
      className="meeting-discussion-item-table-body"
      id="all-meeting-discussion-item-table-body-id"
    >
      <tr
        key="10101"
      >
        <td>
          10101
        </td>
        <td />
        <td />
        <td />
      </tr>
      <tr
        key="10102"
      >
        <td>
          10102
        </td>
        <td />
        <td />
        <td />
      </tr>
    </tbody>
  </table>
</div>
`;
